payroll-processor-functions/src/payroll-processor-functions/Features/Payrolls/PayrollEntity.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using Microsoft.WindowsAzure.Storage.Table;
+
+namespace PayrollProcessor.Functions.Features.Payrolls
+{
+    public class PayrollEntity : TableEntity
+    {
+        public DateTimeOffset CheckDate { get; set; }
+        public Guid EmployeeId { get; set; }
+        public double GrossPayroll { get; set; }
+        public string PayrollPeriod { get; set; } = "";
+        public string EmployeeDepartment { get; set; } = "";
+
+        public static class Map
+        {
+            public static Payroll To(PayrollEntity entity)
+            {
+                return new Payroll(Guid.Parse(entity.RowKey))
+                {
+                    CheckDate = entity.CheckDate,
+                    EmployeeId = entity.EmployeeId,
+                    GrossPayroll = entity.GrossPayroll,
+                    PayrollPeriod = entity.PayrollPeriod,
+                    EmployeeDepartment = entity.EmployeeDepartment
+                };
+            }
+
+            public static PayrollEntity From(Payroll payroll)
+            {
+                return new PayrollEntity
+                {
+                    PartitionKey = payroll.CheckDate.ToString("yyyyMMdd"),
+                    RowKey = payroll.Id.ToString("n"),
+                    CheckDate = payroll.CheckDate,
+                    EmployeeId = payroll.EmployeeId,
+                    GrossPayroll = payroll.GrossPayroll,
+                    PayrollPeriod = payroll.PayrollPeriod,
+                    EmployeeDepartment = payroll.EmployeeDepartment
+                };
+            }
+        }
+    }
+
+    public class EmployeePayrollEntity : TableEntity
```

Comment:
This entity exists to make it easier to query for Payrolls by Employee, whereas the `PayrollEntity` is used for finding Payrolls by date entered.

It's effectively the same data but separated into 2 tables to make querying possible. This is required because Table storage only allows for 1 indexed set of data (Partition Key + Row Key).

Here the Partition Key is the EmployeeId and the Row Key is the Payroll Id

Reply:
At some point it would be useful to make a db schema diagram, with the tables and their partition keys for reference.